lib/spack/spack/ci/common.py

```diff
@@ -49,7 +70,14 @@ def copy_files_to_artifacts(src, artifacts_dir):
         artifacts_dir (str): the destination directory
     """
     try:
-        fs.copy(src, artifacts_dir)
+        if GZIP_SUPPORTED and compress_artifacts and not is_gzipped(src):
+            # Compress and copy in one step
+            src_name = os.path.basename(src)
+            zipped = os.path.join(artifacts_dir, f"{src_name}.gz")
+            with open(src, "rb") as fin, gzip.open(zipped, "wb") as fout:
+                fout.writelines(fin)
```

Comment:
shutil.copyfileobj

Reply:
it seems to? I opened stuff from the archive, but now re-reading the docs on the method I don't know how. Changing it because I agree, shutil is clearer here.

Reply:
yes, unclear if it appends a newline character or not; copyfileobj is verbatim.